src/domain/Client/HttpClientWrapper.cs

```diff
@@ -22,7 +22,9 @@ public async Task<HttpResponseMessage> GetAsync(Uri queryUri)
         {
             try
             {
-                return await wrapped.GetAsync(queryUri);
+                var result = await wrapped.GetAsync(queryUri);
+                result.EnsureSuccessStatusCode();
```

Comment:
This basically throws an exception, yes?  Which is caught on line 29.  I sort of agree with the first answer [here ](https://stackoverflow.com/questions/21097730/usage-of-ensuresuccessstatuscode-and-handling-of-httprequestexception-it-throws) (i.e. don't do this).  Catching exceptions are really computationally expensive (even try/catch  on it's own is a lot of code), so better to use `IsSuccessStatusCode` IMO